contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,621 @@
+                accJoePerShare: 0,
+                rewarder: _rewarder,
+                totalVeJoe: 0,
+                totalBoostedAmount: 0
+            })
+        );
+        lpTokens.add(address(_lpToken));
+        emit Add(poolInfo.length.sub(1), allocPoint, _lpToken, _rewarder);
+    }
+
+    /// @notice Update the given pool's JOE allocation point and `IRewarder` contract. Can only be called by the owner.
+    /// @param _pid The index of the pool. See `poolInfo`.
+    /// @param _allocPoint New AP of the pool.
+    /// @param _rewarder Address of the rewarder delegate.
+    /// @param overwrite True if _rewarder should be `set`. Otherwise `_rewarder` is ignored.
+    function set(
+        uint256 _pid,
+        uint256 _allocPoint,
+        IRewarder _rewarder,
+        bool overwrite
+    ) external onlyOwner {
+        PoolInfo memory pool = poolInfo[_pid];
+        totalAllocPoint = totalAllocPoint.sub(poolInfo[_pid].allocPoint).add(_allocPoint);
+        pool.allocPoint = _allocPoint;
+        if (overwrite) {
+            _rewarder.onJoeReward(address(0), 0); // sanity check
+            pool.rewarder = _rewarder;
+        }
+        poolInfo[_pid] = pool;
+        emit Set(_pid, _allocPoint, overwrite ? _rewarder : pool.rewarder, overwrite);
+    }
+
+    /// @notice View function to see pending JOE on frontend.
+    /// @param _pid The index of the pool. See `poolInfo`.
+    /// @param _user Address of user.
+    /// @return pendingJoe JOE reward for a given user.
+    //          bonusTokenAddress The address of the bonus reward.
+    //          bonusTokenSymbol The symbol of the bonus token.
+    //          pendingBonusToken The amount of bonus rewards pending.
+    function pendingTokens(uint256 _pid, address _user)
+        external
+        view
+        returns (
+            uint256 pendingJoe,
+            address bonusTokenAddress,
+            string memory bonusTokenSymbol,
+            uint256 pendingBonusToken
+        )
+    {
+        PoolInfo memory pool = poolInfo[_pid];
+        UserInfo storage user = userInfo[_pid][_user];
+        uint256 accJoePerShare = pool.accJoePerShare;
+
+        if (block.timestamp > pool.lastRewardTimestamp && pool.totalBoostedAmount != 0) {
+            uint256 secondsElapsed = block.timestamp.sub(pool.lastRewardTimestamp);
+            uint256 joeReward = secondsElapsed.mul(joePerSec()).mul(pool.allocPoint).div(totalAllocPoint);
+            accJoePerShare = accJoePerShare.add(joeReward.mul(ACC_TOKEN_PRECISION).div(pool.totalBoostedAmount));
+        }
+
+        // TODO: Precision.
+        uint256 userBoost = 1000;
+        uint256 userVeJoeBalance = VEJOE.balanceOf(_user);
+        if (userVeJoeBalance > 0) {
+            userBoost = userBoost.add(userVeJoeBalance.mul(maxBoostFactor).div(pool.totalVeJoe));
+        }
+        pendingJoe = user
+            .amount
+            .mul(userBoost)
+            .mul(accJoePerShare)
+            .div(1000)
```

Comment:
is this `1000` coming from `userBoost`, if so let's replace this with `userBoost` instead of hardcoding the `1000`

Reply:
I've moved this into being needed in just one function (I don't think I need to make it a named variable now). It is there because the `maxBoostFactor` is 1500 (which represents 1.5%).